controllers/services/users.js

```diff
@@ -33,115 +35,108 @@ module.exports = function(app) {
-      return res.redirect('/login');
+      res.redirect('/login');
     }
 
+    var absUsersPath = path.resolve(__dirname + '/../../views/_partials/users.ejs');
     request({
-      url: `${SERVICES_URL}/users/paid`,
-      method: "POST",
+      url: `${SERVICES_URL}/users/` + req.params.netid + `/paid`,
+      method: "PUT",
       headers: {
-        "Authorization": GROOT_ACCESS_TOKEN
-      },
-      body: {
-        "token" : req.session.student.token,
-        "netid" : req.params["approvedUserNetID"],
+        "Authorization": GROOT_ACCESS_TOKEN,
+        "Netid": req.session.student.netid
       },
       json: true
-    }, function(err) {
-      if(err) {
-        req.flash('error', "There was an issue, and the member may not have been added. Please contact someone from the Admin committee.");
+    }, function(err, response, body) {
+      if (response && response.statusCode == 200 && body) {
+        res.status(200).send(ejs.render("<%- include('" + absUsersPath + "') %>", { users : body.data } ));
       } else {
-        req.flash('success', "The member was added successfully!");
+        res.status(response.statusCode).send(body.error);
       }
+    });
+  });
 
-      res.redirect('/intranet/users');
+  app.delete('/intranet/users/:netid', function(req, res) {
+    if(!req.session.roles.isAdmin && !req.session.roles.isTop4) {
+      res.redirect('/login');
+    }
+
+    var absUsersPath = path.resolve(__dirname + '/../../views/_partials/users.ejs');
+    request({
+      url: `${SERVICES_URL}/users/` + req.params.netid,
+      method: "DELETE",
+      headers: {
+        "Authorization": GROOT_ACCESS_TOKEN,
+        "Netid": req.session.student.netid
+      },
+      json: true,
+      body: {}
+    }, function(error, response, body) {
+      if (response && response.statusCode == 200 && body) {
+        res.status(200).send(ejs.render("<%- include('" + absUsersPath + "') %>", { users : body.data, me: req.session.student } ));
+      } else {
+        res.status(response.statusCode).send(body.error);
+      }
     });
   });
 
   app.post('/join', function(req, res) {
-    var userData = {
-      first_name: req.body.first_name,
-      last_name: req.body.last_name,
-      netid: req.body.netid,
-      uin: req.body.uin
-    };
     request({
-      url: `${SERVICES_URL}/users/newUser`,
+      url: `${SERVICES_URL}/users`,
       method: "POST",
       headers: {
         "Authorization": GROOT_ACCESS_TOKEN
       },
-      body: userData,
+      body: req.body,
       json: true
     }, function(err, response, body) {
       if(err || !response || response.statusCode != 200) {
-        req.flash('error', err || body.error);
+        req.flash('error', (body && body.error) || err);
       } else {
-        req.flash('success', "Added as a premember");
+        req.flash('success', body.message);
       }
       res.redirect('/join');
     });
   });
 
   app.post('/login', function(req, res){
```

Comment:
So one cool thing that affects development is that we no longer have to connect to the production database on the acm server to login to the website. Now, in development, the users service talks to crowd with the credentials and basically does a find or create User (fetching additional info like their name automatically) and returning it along with the session token.